### PR TITLE
Fix outdated client-retrofit.rst

### PR DESCRIPTION
**Pull request by trustin**
.. also remove unused fields and parameters from ArmeriaRetrofitBuilder
and ArmeriaCallFactory

### CODE REVIEW

File: site/src/sphinx/client-retrofit.rst [hunk 1]
```diff
@@ -43,12 +43,11 @@ The integration is done by creating an ``HttpClient`` that connects to the desir
...
-
-    Retrofit retrofit = new ArmeriaRetrofitBuilder(httpClient)
-                                       .addConverterFactory(JacksonConverterFactory.create())
-                                       .addCallAdapterFactory(Java8CallAdapterFactory.create())
-                                       .build();
+    Retrofit retrofit = new ArmeriaRetrofitBuilder()
+            .baseUrl("http://localhost:8080/")
+            .addConverterFactory(JacksonConverterFactory.create())
+            .addCallAdapterFactory(Java8CallAdapterFactory.create())
+            .build();
```
Comment (trustin) [inline site/src/sphinx/client-retrofit.rst]: The API became much nicer. Thanks, @kojilin !

File: retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaRetrofitBuilder.java [hunk 2]
```diff
@@ -71,15 +70,13 @@
...
 
-    private static final Pattern GROUP_PREFIX_PATTERN = Pattern.compile("^[_0-9a-z]+$");
     private static final BiFunction<String, ? super ClientOptionsBuilder, ClientOptionsBuilder>
             DEFAULT_CONFIGURATOR = (url, optionsBuilder) -> optionsBuilder;
     private static final String SLASH = "/";
 
     private final Retrofit.Builder retrofitBuilder;
     private final ClientFactory clientFactory;
     private String baseUrl;
-    private String basePath;
```
Comment (kojilin) [inline retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaRetrofitBuilder.java]: oops thx ^^;;;

**Comment by trustin** [comment]
Thanks, @kojilin!

File: retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaRetrofitBuilder.java (hunk 2 above)
Comment (trustin) [inline retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaRetrofitBuilder.java]: No worries. I'd like to recommend [importing our IntelliJ IDEA inspection profile](https://github.com/line/armeria/blob/master/CONTRIBUTING.md#intellij-idea) and go through all the inspection warnings by [pressing F2](https://www.jetbrains.com/help/idea/2017.1/navigating-to-next-previous-error.html).

**Pull request merged**